addons/mail/models/mail_message.py

```diff
@@ -1118,6 +1123,12 @@ def _get_store_partner_name_fields(self):
         self.ensure_one()
         return ["name"]
 
+    def _get_store_attachment_fields(self, target):
+        self.ensure_one()
+        if target.is_current_user(self.env) and self.is_current_user_or_guest_author:
+            return self.env["ir.attachment"]._get_store_attachment_ownership_fields()
```

Comment:
Can't you always add it and then check if it's empty? Otherwise if the heuristic changes for the token you have to change it here too (could be cool to let the store handle not sending the key if empty, if not already done and there isn't a reason to keep the distinction)

Reply:
We always need to send empty values, because the transition from non-empty to empty must clear the value in JS. If we don't send empty values we can never clear them.

Store already optimizes for empty stuff whenever possible, such as adding an empty recordset data not doing anything (in top level, again if it's on a relation empty must be sent instead).